doc/userguide/src/CreatingTestData/AdvancedFeatures.rst

```diff
@@ -399,6 +399,103 @@ lower limit, upper limit and step.
+   ================= ================ ============ ========= ========= ============= =======
+
+The above test case will log `0,a,b` and `1,c,d`.
+
+For In Enumerate loops are new in Robot 2.9.
+
+For in Zip
+~~~~~~~~~~
+
+Some tests build up several related lists, then loop over them together.
+Robot Framework has a shortcut for this case: `FOR ... IN ZIP ...`, which
+is derived from the
+`Python built-in zip function <https://docs.python.org/2/library/functions.html#zip>`.
+
+This may be easiest to show with an example:
+
+.. table:: For in zip examples
+   :class: example
+   ================ ======================= ======================= =================== ================= =========== ========
+   Test Case        Action                  Argument                Argument            arg               arg
+   ================ ======================= ======================= =================== ================= =========== ========
+   Flat             [Documentation]         Doing it without any    loops at all
+   \                Number Should Be Named  ${1}                    one
+   \                Number Should Be Named  ${2}                    two
+   \                Number Should Be Named  ${5}                    five
+   Old Style        [Documentation]         Before Robot 2.9
+   \                ${numbers}=             Create List             ${1}                ${2}              ${5}
+   \                ${names}=               Create List             one                 two               five
+   \                ${length}=              Get Length              ${numbers}
+   \                :FOR                    ${idx}                  IN RANGE            ${length}
+   \                                        Number Should Be Named  ${numbers}[${idx}]  ${names}[${idx}]
+   With Zip         [Documentation]         Robot 2.9               and after
+   \                ${numbers}=             Create List             ${1}                ${2}              ${5}
+   \                ${names}=               Create List             one                 two               five
+   \                :FOR                    ${number}               ${name}             IN ZIP            ${numbers}  ${names}
+   \                                        Number Should Be Named  ${number}           ${name}
+   ================ ======================= ======================= =================== ================= =========== ========
+
+Like `IN RANGE` and `IN ENUMERATE` loops, this type of loop requires
+the `IN` cell to read `IN ZIP`.
+
+Note that any lists or other iterables appearing after `IN ZIP` should usually be
+prefixed with `$` instead of `@` because For-In-Zip needs to keep track of each
+list separately.
+
+`IN ZIP` loops require the same number of loop variables as lists to loop over,
+and will stop after the shortest list is complete.
+
+For In Zip loops are new in Robot 2.9, and are named after a Python built-in function.
```

Comment:
`Robot 2.9` -> `Robot Framework 2.9`. The latter part of the sentence is already explained above.